web_src/js/index.js

```diff
@@ -2374,6 +2374,17 @@ $(document).ready(async () => {
     });
   });
 
+  $('.pinned-repo-btn').on('click', function (e) {
+    const $this = $(this);
+    e.preventDefault();
+    $.post($this.data('url'), {
+      _csrf: csrf,
+      name: $this.data('name'),
+      status: $this.data('status'),
+      is_btn: true,
+    }).done(reload());
```

Comment:
We could return the new state of the button at the end of this and just update the button.... It seems strange to go to the effort of using a javascript to replace a form when you're not going to prevent a reload.

Reply:
Indeed, unless the changes after the POST are significant work (button update is not), just update the DOM in JS without a reload, it's a much better user experience.

Reply:
> Indeed, unless the changes after the POST are significant work (button update is not), just update the DOM in JS without a reload, it's a much better user experience.

But It will not only change the button, but also will change the pinned repos list ...

Reply:
I guess it's fine but sub-optimal. I know that in other parts of the UI we sometimes respond with HTML in the POST which is then replaced in JS, maybe that is an option.